registry/coder-labs/modules/copilot-cli/README.md

```diff
@@ -0,0 +1,213 @@
+---
+display_name: Copilot CLI
+description: GitHub Copilot CLI agent for AI-powered terminal assistance
+icon: ../../../../.icons/github.svg
+verified: false
+tags: [agent, copilot, ai, github, cli, tasks]
+---
+
+# Copilot CLI
+
+Run [GitHub Copilot CLI](https://docs.github.com/copilot/concepts/agents/about-copilot-cli) in your workspace for AI-powered coding assistance directly from the terminal. This module integrates with [AgentAPI](https://github.com/coder/agentapi) for task reporting in the Coder UI.
+
+```tf
+module "copilot_cli" {
+  source   = "registry.coder.com/coder-labs/copilot-cli/coder"
+  version  = "0.1.0"
+  agent_id = coder_agent.example.id
+  workdir  = "/home/coder/project"
+}
+```
+
+> [!NOTE]
+> By default, this module is configured to run the embedded chat interface as a path-based application. In production, we recommend that you configure a [wildcard access URL](https://coder.com/docs/admin/setup#wildcard-access-url) and set `subdomain = true`. See [here](https://coder.com/docs/tutorials/best-practices/security-best-practices#disable-path-based-apps) for more details.
+
+## Prerequisites
+
+- **Node.js v22+** and **npm v10+**
+- **Active Copilot subscription** (GitHub Copilot Pro, Pro+, Business, or Enterprise)
+- **GitHub authentication** via one of:
+  - Direct token via `github_token` variable
+  - Coder external authentication (recommended)
+  - Interactive login in Copilot CLI
+
+## Examples
+
+### Usage with Tasks
+
+For development environments where you want Copilot CLI to have full access to tools and automatically resume sessions:
+
+```tf
+data "coder_parameter" "ai_prompt" {
+  type        = "string"
+  name        = "AI Prompt"
+  default     = ""
+  description = "Initial task prompt for Copilot CLI."
+  mutable     = true
+}
+
+module "copilot_cli" {
+  source   = "registry.coder.com/coder-labs/copilot-cli/coder"
+  version  = "0.1.0"
+  agent_id = coder_agent.example.id
+  workdir  = "/home/coder/project"
+
+  github_token    = "your_github_token_here" # Or use data.coder_external_auth.github.access_token
```

Comment:
let's replace `your_github_token_here` with a `var.github_token` and explicitly repurpose this example if someone wants to use a PAT instead of the Coder-provided OAuth-based access token. 

For Coder Tasks, we should recommend passing in the external auth name and using that exclusively.